### PR TITLE
Merged DbConnectionClosed, DbConnectionFactory, SqlAuthenticationProviderManager

### PR DESCRIPTION
**Pull request by edwardneal**
Contributes to #1261.

This merges the three classes above.  DbConnectionFactory has a few references to the metric reporting processes (which are different between .NET Core and Framework) and I've hidden these behind conditional compilation.

### CODE REVIEW

**Review by David-Engel — APPROVED** [review]

**Comment by DavoudEshtehari** [comment]
@edwardneal Can you address the conflicts, please?

**Comment by edwardneal** [comment]
Thanks @DavoudEshtehari, I've addressed the conflicts. CI is effectively done - there are a few test failures, which seem to be unrelated.

**Comment by DavoudEshtehari** [comment]
One more conflict after merging your PR. ;)

**Comment by edwardneal** [comment]
Thanks - re-merged. The test failures remain unrelated

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/ProviderBase/DbConnectionFactory.cs [hunk 1]
```diff
@@ -412,7 +443,7 @@ private DbConnectionPool GetConnectionPool(DbConnection owningObject, DbConnecti
 
         internal DbConnectionPoolGroup GetConnectionPoolGroup(DbConnectionPoolKey key, DbConnectionPoolGroupOptions poolOptions, ref DbConnectionOptions userConnectionOptions)
```
[the remark this answers is not in the document]
Comment (edwardneal) [reply src/Microsoft.Data.SqlClient/src/Microsoft/Data/ProviderBase/DbConnectionFactory.cs:443]: I understand, but can only see one example - `_performanceCounters` wasn't grouped with the conditional block a dozen lines below. I've moved this, and also removed another redundant conditional block which was checking for Windows 9x support on . All of the other members I can see are used in both frameworks - Visual Studio shows references to every common function, property and field for every compilation target. Do you have any others in mind?

The rest of the conditional blocks all revolve around metrics. I think the OTel metrics work (#2211) will eventually remove these.

[the remark this answers is not in the document]
Comment (DavoudEshtehari) [reply src/Microsoft.Data.SqlClient/src/Microsoft/Data/ProviderBase/DbConnectionFactory.cs:443]: Nice, I rechecked and I believe this would be adequate.

**Review by DavoudEshtehari — APPROVED** [review]

**Pull request merged**